protocol/protocol.py

```diff
@@ -129,7 +129,7 @@ def file_upload(filename,dest_dir_url,header_if_match=None,checksum=None):
         else:
             headers['OC-Checksum'] = checksum
 
-    r = client.PUT(filename,os.path.join(dest_dir_url,os.path.basename(filename)),headers)
+    r = client.PUT(filename,dest_dir_url+"/"+os.path.basename(filename),headers)
```

Comment:
Why is this needed? This argument is a URL which needs "/" as separator independently of the OS.